model/models.py

```diff
@@ -128,3 +128,35 @@ def forward(self, sub, rel):
 
 		score = torch.sigmoid(x)
 		return score
+
+class CompGCN_ConvKB(CompGCNBase):
+	'''Implemtents the ConvKB Scoring Function'''
+	def __init__(self, edge_index, edge_type, params=None):
+		super(self.__class__, self).__init__(edge_index, edge_type, params.num_rel, params)
+		self.drop = torch.nn.Dropout(self.p.hid_drop)
+		self.relu = torch.nn.ReLU()
+		self.conv2d1 = torch.nn.Conv2d(1, out_channels=self.p.num_filt, kernel_size=(1, 3), stride=1)
+		self.fake_fc_with_conv = torch.nn.Conv2d(self.p.num_filt,out_channels= 1 , kernel_size=(self.p.embed_dim, 1), stride=self.p.embed_dim,bias=False)
+
+		#Prints the number of parameter in the network to the console
+		total_params = sum(
+			param.numel() for param in self.parameters()
+		)
+		print("Total Parameter: " + str(total_params))
+
+	def forward(self, sub,rel):
+		sub_emb, rel_emb, all_ent, _ = self.forward_base(sub, rel, self.drop, self.drop)
+
+		l = len(sub)
+		sub_emb_repeat = sub_emb.unsqueeze(1).expand(-1,self.p.num_ent,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
+		rel_emb_repeat = rel_emb.unsqueeze(1).expand(-1,self.p.num_ent,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
+		all_ent = all_ent.unsqueeze(0).expand(l,-1,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
+													#size of tensor after manipulation, bs= batch_size
+		x = torch.cat([sub_emb_repeat, rel_emb_repeat, all_ent], 1) 	#bs x 3 x num_ent * embed_dim
+		x = x.transpose(1, 2) 						#bs x num_ent * embed_dim x 3
+		x = x.unsqueeze(1)  						#bs x 1 x num_ent * embed_dim x 3
+		conv_out = self.conv2d1(x) 					#bs x num_filt x num_ent * embed_dim x 1
+		conv_out = self.relu(conv_out)					#bs x num_filt x num_ent * embed_dim x 1
+		score = self.fake_fc_with_conv(conv_out) 			#bs x 1 x num_ent x 1
+		score = score.view(l,self.p.num_ent) 				#bs x num_ent
+		return torch.sigmoid(score) 					#bs x num_ent
```

Comment:
What are all the comments on the right here ? If they should explain something, maybe do not use abbreviations

Reply:
they are showing the size of the tensor after each manipulation to help with understanding what is going on, bs is defined num_filt = Number of filters
num_ent = Number of entities
embed_dim = embedding dimension